### PR TITLE
Adopt Cluster API and write controllers

### PR DESCRIPTION
**Pull request by kris-nova**
We're using the new cluster API! 😄 

 - Adopting new API and using `legacy` API as `providerConfig`
 - Adding anonymous cluster interface so we can hack into the rest of kubicorn
 - Writing controllers for node scaling 
 - Build process for controllers
 - Adding controller field to profiles 

https://github.com/kubernetes/kube-deploy/blob/master/cluster-api

### CODE REVIEW

File: cloud/amazon/public/resources/instanceprofile.go [hunk 1]
```diff
@@ -185,10 +186,23 @@ func (r *InstanceProfile) Apply(actual, expected cloud.Resource, immutable *clus
 		logger.Debug("CreateInstanceProfile error: %v", err)
 		if err.(awserr.Error).Code() != iam.ErrCodeEntityAlreadyExistsException {
 			return nil, nil, err
+		}else {
+			profileinput := &iam.GetInstanceProfileInput{
+				InstanceProfileName: &expected.(*InstanceProfile).Name,
+			}
+			outInstanceProfile, err := Sdk.IAM.GetInstanceProfile(profileinput)
```
Comment (nrb) [inline cloud/amazon/public/resources/instanceprofile.go]: +1 to this - I was thinking of submitting a patch fixing this specifically.

Comment (kris-nova) [reply cloud/amazon/public/resources/instanceprofile.go]: please do I will rebase

Comment (kris-nova) [reply cloud/amazon/public/resources/instanceprofile.go]: I just rebased on master but if you want to PR this gopher it

**Comment by stealthybox** [comment]
What is intended for state-stores?
Is `kubicorn apply` now going to sync the state-store into the Cluster-API?
In this case, the Cluster-API would be the authoritative storage.

What about in reverse? ( when Cluster-API state is newer than the state-store )
`kubicorn sync` ?
`kubicorn controller`?
I suppose the newer state store would win, and in the case of a conflict, we would trust the Cluster-API.

**Comment by davidewatson** [comment]
@stealthybox: How can do you determine whether the state store is newer than the Cluster API objects?

@kris-nova: If we intend to allow users to modify Cluster API types directly, then do you think we should have a `k8s` state store for this case?

**Comment by stealthybox** [comment]
@davidewatson probably using timestamps on the objects from the time of commit

**Comment by kris-nova** [comment]
This PR will be abandon for favor of a different approach.

**Comment by webwurst** [comment]
What approach will it be? Just curious.

**Comment by kris-nova** [comment]
Closing in favor of #572

**Pull request closed without merging**